sphinx/apidoc.py

```diff
@@ -116,7 +116,12 @@ def create_package_file(root, master_package, subroot, py_files, opts, subs, is_
         text += '\n'
 
     # build a list of directories that are szvpackages (contain an INITPY file)
-    subs = [sub for sub in subs if path.isfile(path.join(root, sub, INITPY))]
+    # and also checks the INITPY file is not empty, or there are other python
```

Comment:
an empty INITPY is still a valid INITPY making the directory a package as opposed to namespace

Reply:
True. But before this patch there would be a reference to that package, although the package .rst file would never be created. It seems to have been the behaviour of apidoc.py since many years to exclude those packages.

See:
https://github.com/sphinx-doc/sphinx/blob/87cebf1a070a2c8956ef2939f43d53bf826600e9/sphinx/apidoc.py#L159-L169

That's why I'm reluctant to change an existing deliberate design, but tried to fix the inconsistency. If we change the behaviour in apidoc.py to not skip empty __init__.py files in shall_skip it'd still be consistent. @birkenfeld ?

Reply:
Sounds good.